File: backend/internal/models/guide_application.go
package models

import "time"

type ApplicationStatus string

const (
	ApplicationStatusPending   ApplicationStatus = "pending"
	ApplicationStatusApproved  ApplicationStatus = "approved"
	ApplicationStatusRejected  ApplicationStatus = "rejected"
	ApplicationStatusCancelled ApplicationStatus = "cancelled"
)

type GuideApplication struct {
	ID              string            `json:"id"`
	UserID          string            `json:"user_id"`
	City            string            `json:"city"`
	Status          ApplicationStatus `json:"status"`
	Note            *string           `json:"note,omitempty"`
	ReviewedBy      *string           `json:"reviewed_by,omitempty"`
	ReviewedAt      *time.Time        `json:"reviewed_at,omitempty"`
	TermsAcceptedAt *time.Time        `json:"terms_accepted_at,omitempty"`
	UserName        *string           `json:"user_name,omitempty"`
	UserEmail       *string           `json:"user_email,omitempty"`
	CreatedAt       time.Time         `json:"created_at"`
}

File: mobile/lib/core/models/guide_application.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'guide_application.freezed.dart';
part 'guide_application.g.dart';

@freezed
abstract class GuideApplication with _$GuideApplication {
  const factory GuideApplication({
    required String id,
    @JsonKey(name: 'user_id') required String userId,
    @Default('pending') String status,
    String? note,
    @JsonKey(name: 'reviewed_by') String? reviewedBy,
    @JsonKey(name: 'reviewed_at') DateTime? reviewedAt,
    @JsonKey(name: 'user_name') String? userName,
    @JsonKey(name: 'user_email') String? userEmail,
    @JsonKey(name: 'created_at') DateTime? createdAt,
  }) = _GuideApplication;

  const GuideApplication._();

  factory GuideApplication.fromJson(Map<String, dynamic> json) =>
      _$GuideApplicationFromJson(json);

  bool get isPending => status == 'pending';
  bool get isApproved => status == 'approved';
  bool get isRejected => status == 'rejected';
}
